src/LodeRunner.Data/Services/BaseService.cs

```diff
@@ -95,13 +105,62 @@ public virtual async Task<int> GetCountAsync<TEntity>()
         /// <summary>
         /// Deletes the specified identifier.
         /// </summary>
-        /// <typeparam name="TEntity">The type of the entity.</typeparam>
         /// <param name="id">The identifier.</param>
         /// <returns>The corresponding entity.</returns>
-        public virtual async Task<TEntity> Delete<TEntity>(string id)
+        public virtual async Task<TEntity> Delete(string id)
         {
             EntityType entityType = typeof(TEntity).Name.As<EntityType>();
             return await this.CosmosDBRepository.DeleteDocumentAsync<TEntity>(id, entityType.ToString());
         }
+
+        /// <summary>
+        /// Posts the specified entity.
+        /// </summary>
+        /// <param name="model">The model.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The Created or Updated Entity.</returns>
+        public async Task<TEntity> Save(TEntity model, CancellationToken cancellationToken)
+        {
+           // var returnValue = new Task<TEntity>(() => null);
+            if (model != null && !cancellationToken.IsCancellationRequested)
+            {
+                // Update Entity if CosmosDB connection is ready and the object is valid
+                if (this.CosmosDBRepository.IsCosmosDBReady().Result && this.Validator.ValidateEntity(model))
+                {
+                    return await this.CosmosDBRepository.UpsertDocumentAsync(model, cancellationToken);
+                }
+                else
+                {
+                    // TODO: log specific case scenario, even if IsCosmosDBReady() already will do its own logging.
+
+                    // TODO: log validation errors is any  if not  this.validator.IsValid => this.validator.ErrorMessage
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Creates the API response.
+        /// </summary>
+        /// <param name="returnValue">The return value.</param>
+        /// <returns>Api Response.</returns>
+        protected ApiResponse<TEntity> CreateApiResponse(TEntity returnValue)
+        {
+            ApiResponse<TEntity> result = new ();
+
+            if (!this.Validator.IsValid)
+            {
+                result.Errors = this.Validator.ErrorMessage;
+                result.StatusCode = HttpStatusCode.BadRequest;
+            }
+            else if (returnValue != null)
+            {
+                result.Model = returnValue;
+                result.StatusCode = HttpStatusCode.OK;
+            }
+
```

Comment:
Would a null value pass the `this.Validator.IsValid` check?  It seems like it could for no content responses like a delete operation.  If so then the IsValid check should pass and return the Ok.  It also seems like we could change this to positive logic as it is either valid and passes the check or not:
```csharp
            if (this.Validator.IsValid)
            {
                result.Model = returnValue;
                result.StatusCode = HttpStatusCode.OK;
            }
            else 
            {
                result.Errors = this.Validator.ErrorMessage;
                result.StatusCode = HttpStatusCode.BadRequest;
            }
```

Reply:
This will be addressed in a separate issue https://github.com/orgs/retaildevcrews/projects/12#card-76493062